src/util/utils.h

```diff
@@ -0,0 +1,16 @@
+
+#ifndef __UTILS_UTIL_H__
+#define __UTILS_UTIL_H__
+
+// Standard math utility macros
+
+// TODO: Consider safer macros here for these operations
```

Comment:
I agree with your concern here. We could run into major problems if anyone attempts to use these macros with different data types.

Reply:
Yeah, online has some gcc-specific extension you can use to solve the problem, but alot of libraries (including Arduino) did it exactly like this classic, so I included them here for now with the warning :)